niceman/interface/tests/test_diff.py

```diff
@@ -0,0 +1,41 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the niceman package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+
+from os.path import join as opj, dirname
+from niceman.cmdline.main import main
+
+import logging
+
+from niceman.utils import swallow_logs
+from niceman.tests.utils import assert_in
+
+xeyes_yaml = opj(dirname(__file__), 'files', 'xeyes.yaml')
+xeyes_no_x11_yaml = opj(dirname(__file__), 'files', 'xeyes_no_x11-apps.yaml')
+
+def test_diff_same():
+    with swallow_logs(new_level=logging.DEBUG) as log:
+        args = ['diff', 
+                '--env', 
+                'niceman/interface/tests/files/xeyes.yaml', 
+                '--req', 
+                'niceman/interface/tests/files/xeyes_no_x11-apps.yaml'
+                ]
+        main(args)
+        assert_in('requirements satisfied', log.lines)
+
+def test_diff_different():
+    with swallow_logs(new_level=logging.DEBUG) as log:
+        args = ['diff', 
+                '--env', 
+                'niceman/interface/tests/files/xeyes_no_x11-apps.yaml', 
+                '--req', 
+                'niceman/interface/tests/files/xeyes.yaml'
+                ]
+        main(args)
+        assert_in('needed packages: 1', log.lines)
```

Comment:
from this test I see that you are aiming for a directionality, i.e. "satisfy" relationship, not just a pure diff (how two specs are different).
I think, to support that mode we should add `--satisfy` flag (later might reconsider for `--mode=(diff|satisfy|...smthelse)`.  But otherwise keep it as `spec1 spec2` and adjust documentation that "later is checked to satisfy previous", which would also scale up for multiple specs (again -- collected a bunch of them and want to see that next ones were always "enhanced" (new stuff added) but not really changed/upgraded -- i.e. that I could still satisfy requirements of the previous environment)

Makes sense?

Reply:
another `smthelse` here will be something like `compatible`: given a number of specs to determine if they are not in conflict. i.e. that they are all just a subset of a larger union of them all. And if union is not possible (e.g. in one Debian package X is version 1 and in the other version 2), show for which components, in which specs there is a difference.